pandas/core/arrays/_mixins.py

```diff
@@ -304,6 +304,68 @@ def __repr__(self) -> str:
     # ------------------------------------------------------------------------
     # __array_function__ methods
 
+    def __array_function__(self, func, types, args, kwargs):
+        for x in types:
+            if not issubclass(x, (np.ndarray, NDArrayBackedExtensionArray)):
+                return NotImplemented
+
+        if not args:
```

Comment:
Can you give a code example of this path?

Reply:
`np.delete(arr=np.arange(5), obj=4)`